src/components/QueryEditor/OpenAIEditor.tsx

```diff
@@ -64,24 +72,52 @@ export const OpenAIEditor: React.FC<RawQueryEditorProps> = (props) => {
   }, [schema, stateSchema]);
 
   const generateQuery = () => {
+    reportInteraction('grafana_ds_adx_openai_query_generated');
     setWaiting(true);
     setError(false);
     setErrorMessage(TOKEN_NOT_FOUND);
-    reportInteraction('grafana_ds_adx_openai_query_generated');
-    datasource
-      .generateQueryForOpenAI(`${baselinePrompt}${prompt}"""`)
-      .then((resp) => {
-        setWaiting(false);
-        setGeneratedQuery(resp);
-        onRawQueryChange(resp);
+    if (enabled) {
+      newGenerateQuery(); //eventually this will replace this function
+    } else {
+      datasource
```

Comment:
Will this still work without the OpenAI config?

Reply:
Shoot, I thought it would work since the key was already saved but you're correct now that I double-check it. I'll add the config back but disable the field or something.

Reply:
I was right and the API being down was the problem :sweat_smile: . If users have configured an API token previously the removal of the Config component won't prohibit them from using it.

Reply:
:sob: OpenAI had an outage today -- we'd just deployed a new feature to ops and were testing, totally threw us off, too